tests/test_arthur.py

```diff
@@ -61,7 +62,54 @@ def setUp(self):
     def tearDown(self):
         self.conn.flushdb()
 
+    def test_add_task(self):
+        """Check whether tasks are added"""
+
+        app = Arthur(self.conn, async_mode=False)
+
+        initial_tasks = len(app._tasks.tasks)
+        app.add_task("arthur.task", "backend", {"a": "a", "b": "b"})
+        after_tasks = len(app._tasks.tasks)
+
+        self.assertGreater(after_tasks, initial_tasks)
+
+    def test_add_duplicated_task(self):
+        """Check whether an exception is thrown when a duplicated task is added"""
+
+        app = Arthur(self.conn, async_mode=False)
+
+        app.add_task("arthur.task", "backend", {"a": "a", "b": "b"})
+
+        with self.assertRaises(AlreadyExistsError):
+            app.add_task("arthur.task", "backend", {"a": "a", "b": "b"})
+
+    def test_remove_task(self):
+        """Check whether the removal of tasks is properly handled"""
+
+        task_1 = "arthur.task-1"
+        task_2 = "arthur.task-2"
+
+        task_params = {"backend": "backend", "backend_params": {"a": "a", "b": "b"}}
+
+        app = Arthur(self.conn, async_mode=False)
+
+        app.add_task(task_1, task_params['backend'], task_params['backend_params'])
+        app.add_task(task_2, task_params['backend'], task_params['backend_params'])
+        tasks = len(app._tasks.tasks)
+
+        self.assertEqual(tasks, 2)
+
+        self.assertTrue(app.remove_task(task_1))
+        self.assertTrue(app.remove_task(task_2))
+
+        tasks = len(app._tasks.tasks)
+        self.assertEqual(tasks, 0)
+
+        self.assertFalse(app.remove_task("task-x"))
```

Comment:
I think you should move this last test to a new test case in a separated function